bookshelf/bookshelf.d.ts

```diff
@@ -177,8 +191,12 @@ declare module 'bookshelf' {
 			keys() : string[];
 			last() : T;
 			lastIndexOf(value : any, fromIndex? : number) : number;
-			map(predicate? : Lodash.ListIterator<T, boolean>|Lodash.DictionaryIterator<T, boolean>|string, thisArg? : any) : T[];
+
+			// See https://github.com/Microsoft/TypeScript/blob/v1.8.10/lib/lib.core.es7.d.ts#L1122
+			map<U>(predicate? : Lodash.ListIterator<T, U>|string, thisArg? : any) : U[];
```

Comment:
The difference is `iter[2]` and `iter['two']`. See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/lodash/lodash-3.10.d.ts#L15807. My only concern is maintaining lodash APIs even though I may not use them. I wish there as a better way to do that.

You are right in your assumption though. I don't know why typescript doesn't work either. My understanding of typescript says that (3) and the last are identical, but if only the last works (or is easier to write), then the last may be the best solution. If there is a reason why lodash splits the methods up instead of combining them like I have, I'm unsure of the why. Perhaps we should split them up identical to the lodash methods, i.e.
```
map<U>(predicate? : Lodash.ListIterator<T, U>, thisArg? : any) : U[];
map<U>(predicate? : Lodash.Dictionary<T, U>, thisArg? : any) : U[];
map<U>(predicate? : string) : U[];
```
(I just realized that a string predicate doesn't have a this parameter). This might be why sometimes I see `map<Book>(...)` instead of `map(...)`.